src/lib/include/x86_vector_triple.h

```diff
@@ -0,0 +1,89 @@
+#ifndef DISTOPIA_X86_VECTOR_TRIPLE
+#define DISTOPIA_X86_VECTOR_TRIPLE
+
+#include "distopia_type_traits.h"
+#include "x86_swizzle.h"
+#include "x86_tgintrin.h"
+#include "x86_vectors.h"
+
+// forward declaration defines default EnableIfX template arguments
+template <typename VectorT, typename ScalarT, EnableIfVector<VectorT> = 0,
+          EnableIfFloating<ScalarT> = 0, EnableIfMatching<VectorT, ScalarT> = 0>
+class InterleavedVectorTriple;
+
+// forward declaration defines default EnableIfX template arguments
+template <typename VectorT, typename ScalarT, EnableIfVector<VectorT> = 0,
+          EnableIfFloating<ScalarT> = 0, EnableIfMatching<VectorT, ScalarT> = 0>
+class DeinterleavedVectorTriple;
+
+template <typename VectorT, typename ScalarT, EnableIfVector<VectorT> = 0,
+          EnableIfFloating<ScalarT> = 0, EnableIfMatching<VectorT, ScalarT> = 0>
+class VectorTriple {
+public:
+  VectorT a;
+  VectorT b;
+  VectorT c;
+  constexpr static std::size_t nvals_per_pack = ValuesPerPack<VectorT>::value;
+
+  // from 3 SIMD Vector datatypes eg __m128 or __m128d
+  inline explicit VectorTriple(VectorT a, VectorT b, VectorT c) : a(a), b(b), c(c) {}
+
+  // load from a vector of ScalarT eg float* or double * as constructor
+  inline explicit VectorTriple(ScalarT *source) {
+    a = loadu_p<VectorT>(source);
+    b = loadu_p<VectorT>(source + nvals_per_pack);
+    c = loadu_p<VectorT>(source + 2 * nvals_per_pack);
+  }
+
+  // load from a vector of ScalarT eg float* or double *
+  inline void load(ScalarT *source) {
+    a = loadu_p<VectorT>(source);
+    b = loadu_p<VectorT>(source + nvals_per_pack);
+    c = loadu_p<VectorT>(source + 2 * nvals_per_pack);
+  }
+
+  // to a vector of ScalarT eg float* or double *
+  inline void store(ScalarT *target) {
+    storeu_p(target, a);
+    storeu_p(target[nvals_per_pack], b);
+    storeu_p(target[2 * nvals_per_pack], c);
+  }
+};
+
+// This has been forward declared, so no longer allows EnableIfX = 0 default
+// declaration in same scope
+template <typename VectorT, typename ScalarT, EnableIfVector<VectorT>,
+          EnableIfFloating<ScalarT>, EnableIfMatching<VectorT, ScalarT>>
+class InterleavedVectorTriple : public VectorTriple<VectorT, ScalarT> {
+public:
+  //inherit both constructors with right SFINAE template arguments?
```

Comment:
@ngbl is this the right way to inherit a constructor with our EnableIf SFINAE? it seems to compile and work fine. As I understand it the only templates that can actually be compiled are of the form:
```c++
VectorTriple<SIMD,MatchingFloat,0,0,0>
```
:)